src/RadioButtonCatalog.js

```diff
@@ -19,72 +20,69 @@ const RadioButtonCatalog = () => {
 }
 
 class RadioButtonHero extends Component {
-
-  ripples = [];
-  initRipple = buttonEl => buttonEl && this.ripples.push(new MDCRipple(buttonEl));
-
-  componentWillUnmount() {
-    this.ripples.forEach(ripple => ripple.destroy());
-  }
+  radios = [];
```

Comment:
What purpose does the `radios` array serve here if the `Radio` component handles its own lifecycle?

If it doesn't have any purpose, you could remove it and the `refCallback`s from the `Radio` components inside the `render` method and turn this into a purely functional component.

Reply:
Oops, thanks for catching! I actually want Radio's parent to handle its lifecycle, because the `MDCFormField` object needs to be able to reference the `MDCRadio` object.

Reply:
And I made Radio a functional component so FormFieldRadio can have the `MDCRadio` object. This means that RadioHero will also have `MDCRadio` objects. Let me know if there's a better way of doing this.